p2p.go

```diff
@@ -328,22 +331,27 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
 		}),
 		libp2p.Identity(prikey),
 		libp2p.DefaultSecurity,
-		libp2p.Transport(func(upgrader *tptu.Upgrader) *tcp.TcpTransport {
-			return &tcp.TcpTransport{Upgrader: upgrader, ConnectTimeout: cfg.ConnectTimeout}
+		libp2p.Transport(func(upgrader transport.Upgrader) *tcp.TcpTransport {
+			tp, err := tcp.NewTCPTransport(upgrader, &network.NullResourceManager{}, tcp.WithConnectionTimeout(cfg.ConnectTimeout))
+			if err != nil {
+				Logger().Panic("failed to new tcp transport", zap.Error(err))
+			}
+			return tp
 		}),
 		libp2p.Muxer("/yamux/2.0.0", yamux.DefaultTransport),
-		libp2p.ConnectionManager(connmgr.NewConnManager(cfg.ConnLowWater, cfg.ConnHighWater, cfg.ConnGracePeriod)),
+		libp2p.ConnectionManager(connMgr),
+		libp2p.DefaultPrometheusRegisterer,
 	}
 
 	if !cfg.SecureIO {
 		opts = append(opts, libp2p.NoSecurity)
 	} else {
-		opts = append(opts, libp2p.Security(secio.ID, secio.New))
+		// opts = append(opts, libp2p.Security(secio.ID, secio.New))
```

Comment:
this is not needed?

Reply:
[secio](https://github.com/libp2p/go-libp2p-secio) is already deprecated:
> secio is not maintained any more, and we advise against using it.
Use [go-libp2p-tls](https://github.com/libp2p/go-libp2p-tls/) and [go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise/) instead.

and tls and noise is the two default security of libp2p, so we can just remove it